.circleci/config.yml

```diff
@@ -0,0 +1,35 @@
+version: 2
+jobs:
+  debug_build_and_test:
+    docker:
+      - image: ros:foxy
+    steps:
+      - checkout
+      - run:
+          name: Set Up Container
+          command: |
+            apt update -qq && apt install -y build-essential cmake python3-colcon-common-extensions python3-rosdep libeigen3-dev
+            apt upgrade -y
+            source `find /opt/ros -maxdepth 2 -name local_setup.bash | sort | head -1`
+            mkdir -p src/grid_map && mv `find -maxdepth 1 -not -name . -not -name src` src/grid_map/
+            rosdep update
+            rosdep install -y --ignore-src --from-paths src
+      - run:
+          name: Debug Build
+          command: |
+            source `find /opt/ros -maxdepth 2 -name local_setup.bash | sort | head -1`
+            colcon build --parallel-workers 1
+      - run:
+          name: Run Tests
+          command: |
+            source `find /opt/ros -maxdepth 2 -name local_setup.bash | sort | head -1`
+            colcon test --parallel-workers 1
+            colcon test-result --verbose
+workflows:
+  version: 2
+  ros_build:
+    jobs:
+      - debug_build_and_test
+          filters:
```

Comment:
This isn't the right place for the filters. it should be up in the job definition. See the docs for example.

```
workflows:
  version: 2
  commit:
    jobs:
      - test
      - deploy
  nightly:
    triggers:
      - schedule:
          cron: "0 0 * * *"
          filters:
            branches:
              only:
                - master
                - beta
    jobs:
      - coverage
```

Reply:
Why wouldn't this work? As far as I understand this should build on every commit without a need for a schedule.

That's the example I followed: https://circleci.com/docs/2.0/configuration-reference/#filters-1

Please let me know if I am missing something.

Reply:
I'm not saying schedule. I'm showing you where the filters yaml should go in this arbitrary example that included it.